Submitted By: Archaic <archaic at linuxfromscratch dot org>
Date: 2005-03-22
Initial Package Version: 4.0.7
Origin: http://cvs.pld.org.pl/shadow/libmisc/log.c?r1=1.7&r2=1.8
Upstream Status: Already in CVS
Description: Shadow fails to update the time of last login as noted by lastlog
when logging into a regular console. The Port (tty) column does get updated.
Remote logins and xterm logins seem to be not be broken, just console.

diff -Naur shadow-4.0.7.orig/libmisc/log.c shadow-4.0.7/libmisc/log.c
--- shadow-4.0.7.orig/libmisc/log.c	2003-12-17 12:52:25.000000000 +0000
+++ shadow-4.0.7/libmisc/log.c	2005-03-22 05:53:59.000000000 +0000
@@ -89,6 +89,7 @@
 
 	ll_time = newlog.ll_time;
 	time (&ll_time);
+	newlog.ll_time = ll_time;
 	strncpy (newlog.ll_line, line, sizeof newlog.ll_line);
 #if HAVE_LL_HOST
 	strncpy (newlog.ll_host, host, sizeof newlog.ll_host);
